lib/rex/post/meterpreter/packet_dispatcher.rb

```diff
@@ -639,6 +638,44 @@ def deregister_inbound_handler(handler)
   attr_accessor :receiver_thread # :nodoc:
   attr_accessor :dispatcher_thread # :nodoc:
   attr_accessor :waiters # :nodoc:
+
+  @tlv_log_file = nil
+
+  def log_packet(packet, packet_type)
+    if framework.datastore['sessiontlvlogging']&.downcase == 'console'
+      log_packet_to_console(packet, packet_type)
+    elsif framework.datastore['sessiontlvlogging']&.downcase =~ /^file:.+/
+      log_packet_to_file(packet, packet_type)
+    end
+  end
+
+  def log_packet_to_console(packet, packet_type)
+    if packet_type == :inbound
+      print "\n%bluRECV%clr: #{packet.inspect}\n"
+    elsif packet_type == :outbound
+      print "\n%redSEND%clr: #{packet.inspect}\n"
+    end
+  end
+
+  def log_packet_to_file(packet, packet_type)
+    path = framework.datastore['sessiontlvlogging'].downcase.split('file:').last
```

Comment:
I don't think we can blindly downcase paths here? 🤔 

Relevant reads:

https://betterprogramming.pub/mistakes-ive-made-treating-file-paths-as-strings-ef831eecb33a

https://www.howtogeek.com/137096/6-ways-the-linux-file-system-is-different-from-the-windows-file-system/